File: sdkwork-im-sdk-typescript/composed/src/rtc-module.ts
import { buildJsonRtcSignalRequest } from './builders.js';
import type {
  CreateRtcSessionRequest,
  InviteRtcSessionRequest,
  IssueRtcParticipantCredentialRequest,
  PostJsonRtcSignalOptions,
  PostRtcSignalRequest,
  RtcParticipantCredential,
  RtcRecordingArtifact,
  RtcSession,
  RtcSignalEvent,
  UpdateRtcSessionRequest,
} from './types.js';
import type { ImSdkContext } from './sdk-context.js';

export class ImRtcModule {
  constructor(private readonly context: ImSdkContext) {}

  create(body: CreateRtcSessionRequest): Promise<RtcSession> {
    return this.context.transportClient.rtc.createRtcSession(body);
  }

  invite(
    rtcSessionId: string | number,
    body: InviteRtcSessionRequest,
  ): Promise<RtcSession> {
    return this.context.transportClient.rtc.inviteRtcSession(rtcSessionId, body);
  }

  accept(
    rtcSessionId: string | number,
    body: UpdateRtcSessionRequest,
  ): Promise<RtcSession> {
    return this.context.transportClient.rtc.acceptRtcSession(rtcSessionId, body);
  }

  reject(
    rtcSessionId: string | number,
    body: UpdateRtcSessionRequest,
  ): Promise<RtcSession> {
    return this.context.transportClient.rtc.rejectRtcSession(rtcSessionId, body);
  }

  end(
    rtcSessionId: string | number,
    body: UpdateRtcSessionRequest,
  ): Promise<RtcSession> {
    return this.context.transportClient.rtc.endRtcSession(rtcSessionId, body);
  }

  postSignal(
    rtcSessionId: string | number,
    body: PostRtcSignalRequest,
  ): Promise<RtcSignalEvent> {
    return this.context.transportClient.rtc.postRtcSignal(rtcSessionId, body);
  }

  postJsonSignal(
    rtcSessionId: string | number,
    signalType: string,
    options: PostJsonRtcSignalOptions,
  ): Promise<RtcSignalEvent> {
    return this.postSignal(
      rtcSessionId,
      buildJsonRtcSignalRequest(signalType, options),
    );
  }

  issueParticipantCredential(
    rtcSessionId: string | number,
    body: IssueRtcParticipantCredentialRequest,
  ): Promise<RtcParticipantCredential> {
    return this.context.transportClient.rtc.issueRtcParticipantCredential(
      rtcSessionId,
      body,
    );
  }

  getRecordingArtifact(
    rtcSessionId: string | number,
  ): Promise<RtcRecordingArtifact> {
    return this.context.transportClient.rtc.getRtcRecordingArtifact(rtcSessionId);
  }
}

File: sdkwork-im-sdk-flutter/composed/lib/src/rtc_module.dart
import 'package:im_sdk_generated/im_sdk_generated.dart';

import 'builders.dart';
import 'context.dart';
import 'types.dart';

class ImRtcModule {
  final ImSdkContext context;

  ImRtcModule(this.context);

  Future<RtcSession?> create(CreateRtcSessionRequest body) {
    return context.transportClient.rtc.createRtcSession(body);
  }

  Future<RtcSession?> invite(
    String rtcSessionId,
    InviteRtcSessionRequest body,
  ) {
    return context.transportClient.rtc.inviteRtcSession(rtcSessionId, body);
  }

  Future<RtcSession?> accept(
    String rtcSessionId,
    UpdateRtcSessionRequest body,
  ) {
    return context.transportClient.rtc.acceptRtcSession(rtcSessionId, body);
  }

  Future<RtcSession?> reject(
    String rtcSessionId,
    UpdateRtcSessionRequest body,
  ) {
    return context.transportClient.rtc.rejectRtcSession(rtcSessionId, body);
  }

  Future<RtcSession?> end(
    String rtcSessionId,
    UpdateRtcSessionRequest body,
  ) {
    return context.transportClient.rtc.endRtcSession(rtcSessionId, body);
  }

  Future<RtcSignalEvent?> postSignal(
    String rtcSessionId,
    PostRtcSignalRequest body,
  ) {
    return context.transportClient.rtc.postRtcSignal(rtcSessionId, body);
  }

  Future<RtcSignalEvent?> postJsonSignal(
    String rtcSessionId, {
    required String signalType,
    required ImPostJsonSignalOptions options,
  }) {
    return postSignal(
      rtcSessionId,
      ImBuilders.jsonRtcSignal(
        signalType: signalType,
        options: options,
      ),
    );
  }

  Future<RtcParticipantCredential?> issueParticipantCredential(
    String rtcSessionId,
    IssueRtcParticipantCredentialRequest body,
  ) {
    return context.transportClient.rtc.issueRtcParticipantCredential(
      rtcSessionId,
      body,
    );
  }

  Future<RtcRecordingArtifact?> getRecordingArtifact(String rtcSessionId) {
    return context.transportClient.rtc.getRtcRecordingArtifact(rtcSessionId);
  }
}
